### PR TITLE
Merge conv2

### PR DESCRIPTION
**Pull request by nouiz**
This PR replace gh-2815 as it have a better way to fix it.

### CODE REVIEW

File: theano/sandbox/cuda/basic_ops.py [hunk 1]
```diff
@@ -3299,9 +3299,6 @@ def perform(self, node, inputs, out_):
             # XXX: We could implement and call CudaNdarray.empty(sh) instead.
             out[0] = cuda_ndarray.cuda_ndarray.CudaNdarray.zeros(sh)
 
-    def do_merge(self, node):
-        return False
-
```
Comment (abergeron) [inline theano/sandbox/cuda/basic_ops.py]: That's not ok, we need to explicitly disable the merge for GpuAllocEmpty nodes.

Comment (abergeron) [inline theano/sandbox/cuda/basic_ops.py]: Ignore this comment, I hadn't realized you used a different approach.

**Comment by abergeron** [comment]
It doesn't pass flake8

**Comment by nouiz** [comment]
fixed

**Pull request merged**